### PR TITLE
feat(rest,types): slash command permissions v2

### PR DESCRIPTION
**Pull request by lts20050703**
Closes: #2184
Upstream: discord/discord-api-docs@4a3b20a

- Nuke batchEditApplicationCommandPermissions (https://discord.com/developers/docs/interactions/application-commands#batch-edit-application-command-permissions)
- update JSDocs Comments (like, a lot, probably will revert unnecessary comments changes next commit)
- Add id to upsertApplicationCommands options

### CODE REVIEW

**Review by meister03 — COMMENTED** [review]
Interesting changes, will compare them with the documentation tonight. Everything else looks fine

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: types/discord.ts [hunk 1]
```diff
@@ -1644,37 +1645,37 @@ export interface DiscordInviteStageInstance {
...
-  /** The type of command. By default this is a application command(ChatInput). */
+  /** Type of command, defaults to `ApplicationCommandTypes.ChatInput` */
   type?: ApplicationCommandTypes;
-  /** Unique id of the parent application */
+  /** ID of the parent application */
   application_id: string;
   /** Guild id of the command, if not global */
   guild_id?: string;
-  /** `ApplicationCommandTypes.ChatInput` command names must match the following regex `^[-_\p{L}\p{N}\p{sc=Deva}\p{sc=Thai}]{1,32}$` with the unicode flag set. If there is a lowercase variant of any letters used, you must use those. Characters with no lowercase variants and/or uncased letters are still allowed. `ApplicationCommandTypes.User` and `ApplicationCommandTypes.Message` commands may be mixed case and can include spaces. */
+  /**	Name of command, 1-32 characters */
```
[the remark this answers is not in the document]
Comment (lts20050703) [reply types/discord.ts:1656]: I think it's unnecessary but I will revert that

**Comment by itohatweb** [comment]
Can you rebase this PR to #2238?

**Comment by lts20050703** [comment]
> Can you rebase this PR to #2238?

@itohatweb resolved merge conflicts

**Comment by itohatweb** [comment]
I'M too lazy to review

**Review by itohatweb — APPROVED** [review]
lgtm for now

**Pull request merged**